### DISCUSSION TITLE
Apple federated account authentication

### DISCUSSION THREAD

**Discussion by denysop** [fastlane/fastlane#20750] [General]
Hi team!

I am recently new to Fastlane but I after my research I did not find anything related to the topic neither in official documentation, nor in source code or even web.

In Apple Business Manager there is a possibility to use federated accounts - integration with identity providers by Google or Microsoft. So when you enter your email Apple redirects you for authentication to one of these: Google, MS and your credentials are actually processed by them (not Apple).
Link to Apple docs: https://support.apple.com/guide/apple-business-manager/intro-to-federated-authentication-axmb19317543/web

From the Spaceship source I see that it uses simple pair "email/password" (possibly with 2FA) and there is nothing related to federated flow as far as I understood.
https://github.com/fastlane/fastlane/blob/e874a47c6e2e0e61590a03d3b71e75e5a505d1ce/spaceship/lib/spaceship/client.rb#L410

Also I tried to use spaceship login for the federated account providing email/password pair but it returns "AccessForbidden".

Fastlane team, could you please consider adding support for the federated accounts flow?

**Comment by Valentin-Rault** [comment]
Any news on this? This is still relevant today

**Comment by arnavVandebron** [comment]
Would love some more info on this, our production build pipe-line is currently blocked by this.

**Comment by lluisgerard** [comment]
Is this really not possible yet?

**Reply by atereshkov** [reply]
+1

**Comment by Chris-Corea** [comment]
Bump 😃

**Comment by puruchandraa** [comment]
Still no update here, bump

**Comment by monchote** [comment]
Ongoing discussion here https://github.com/fastlane/fastlane/issues/27261
